contracts/DefaultStrategy.sol

```diff
@@ -522,248 +443,289 @@ contract DefaultStrategy is UUPSOwnableUpgradeable, UsingRegistryUpgradeable, Ma
-     */
-    function getDeprecatedGroupsWithdrawalDistribution(uint256 withdrawal)
-        internal
-        returns (
-            address[] memory deprecatedGroupsWithdrawn,
-            uint256[] memory deprecatedWithdrawalsPerGroup,
-            uint256 numberDeprecatedGroupsWithdrawn,
-            uint256 remainingWithdrawal
-        )
+     * @notice Returns expected stCELO vs actual stCELO for group.
+     * @param group The group.
+     * @return expectedStCelo The stCELO which group should have.
+     * @return actualStCelo The stCELO which group has.
+     */
+    function getExpectedAndActualStCeloForGroup(address group)
+        public
+        view
+        returns (uint256 expectedStCelo, uint256 actualStCelo)
     {
-        remainingWithdrawal = withdrawal;
-        uint256 numberDeprecatedGroups = deprecatedGroups.length();
-        deprecatedGroupsWithdrawn = new address[](numberDeprecatedGroups);
-        deprecatedWithdrawalsPerGroup = new uint256[](numberDeprecatedGroups);
-        numberDeprecatedGroupsWithdrawn = 0;
-
-        for (uint256 i = 0; i < numberDeprecatedGroups; i++) {
-            numberDeprecatedGroupsWithdrawn++;
-            deprecatedGroupsWithdrawn[i] = deprecatedGroups.at(i);
-            uint256 currentStCeloInStrategy = defaultStrategyTotalStCeloVotes[
-                deprecatedGroupsWithdrawn[i]
-            ];
-            uint256 currentVotes = IManager(manager).toCelo(currentStCeloInStrategy);
-            deprecatedWithdrawalsPerGroup[i] = Math.min(remainingWithdrawal, currentVotes);
-            remainingWithdrawal -= deprecatedWithdrawalsPerGroup[i];
-
-            if (currentVotes == deprecatedWithdrawalsPerGroup[i]) {
-                if (!deprecatedGroups.remove(deprecatedGroupsWithdrawn[i])) {
-                    revert FailedToRemoveDeprecatedGroup(deprecatedGroupsWithdrawn[i]);
-                }
-                emit GroupRemoved(deprecatedGroupsWithdrawn[i]);
-            }
+        address head = activeGroups.getHead();
+        expectedStCelo = totalStCeloInDefaultStrategy / activeGroups.getNumElements();
+        if (group == head) {
+            uint256 divisionResidue = totalStCeloInDefaultStrategy -
+                (expectedStCelo * activeGroups.getNumElements());
+            expectedStCelo += divisionResidue;
+        }
 
-            subtractFromStrategyTotalStCeloVotes(
-                deprecatedGroupsWithdrawn[i],
-                currentStCeloInStrategy
-            );
+        actualStCelo = stCELOInGroup[group];
+    }
 
-            if (remainingWithdrawal == 0) {
-                break;
-            }
+    /**
+     * @notice Adds/substracts value to totals of group and
+     * total stCELO in default strategy.
+     * @param strategy The validator group that we are updating.
+     * @param stCeloAmount The amount of stCELO.
+     * @param add Whether to add or substract.
+     */
+    function updateGroupStCelo(
```

Comment:
ditto on visibility

Reply:
I'm using it in unit test to be able to update this manually